pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1285,6 +1288,17 @@ func assignCPUs(ctx *domainContext, config *types.DomainConfig, status *types.Do
 	return nil
 }
 
+// releaseCPUs releases the CPUs that were previously assigned to the VM.
+// The cpumask in the *status* is updated accordingly, and the CPUs are released in the CPUAllocator context.
+func releaseCPUs(ctx *domainContext, config *types.DomainConfig, status *types.DomainStatus) {
```

Comment:
Can this function be called more than once for a given app instance? If so, would the log.Errorf be hit, or will cpuAllocator.Free return nil?
If the former, would it make sense to check if status.VmConfig.CPUs != "" before calling the Free function?

Reply:
The idea is to use it only as a counterpart to `assignCPUs()` in the scope of the `doActivate()` function. So, it should not be called more than once. 
But if someone later decides to use it in another place - yes, it will print an error in case of already freed CPUs (from the CPU Allocator perspective) but the system will continue working.
And yes, I've added a check not to call Free if a domain has an empty CPUs field in the status, thanks!

Reply:
Great